packages/viewer/src/components/scene-tree-table/scene-tree-table.tsx

```diff
@@ -0,0 +1,511 @@
+        (map, c) =>
+          map.set(
+            c,
+            new ElementPool(c, () => this.createColumnCellInstance(c))
+          ),
+        new WeakMap()
+      );
+    }
+  }
+
+  private iterateColumns(
+    f: (
+      column: HTMLVertexSceneTreeTableColumnElement,
+      pool: ElementPool,
+      index: number
+    ) => void
+  ): void {
+    this.columnElements.forEach((column, i) => {
+      const pool = this.stateMap.columnElementPools?.get(column);
+      if (pool != null) {
+        f(column, pool, i);
+      } else {
+        throw new Error('Cannot find pool for column');
+      }
+    });
+  }
+
+  private createHeaderInstance(
+    column: HTMLVertexSceneTreeTableColumnElement
+  ): InstancedTemplate<HTMLElement> | undefined {
+    const template = column.querySelector(
+      'template[slot="header"]'
+    ) as HTMLTemplateElement;
+    if (template != null) {
+      return generateInstanceFromTemplate(template);
+    }
+  }
+
+  private createColumnCellInstance(
+    column: HTMLVertexSceneTreeTableColumnElement
+  ): InstancedTemplate<HTMLElement> {
+    const template = column.querySelector(
+      'template:not([slot="header"])'
+    ) as HTMLTemplateElement;
+    if (template != null) {
+      return generateInstanceFromTemplate(template);
+    } else {
+      throw new Error('Column is missing cell template element');
+    }
+  }
+
+  private computeCellHeight = async (): Promise<void> => {
+    if (this.isComputingCellHeight && this.columnElements.length > 0) {
+      const dummyData: LoadedRow = {
+        index: 0,
+        node: {
+          id: { hex: '' },
+          name: 'Dummy row',
+          expanded: false,
+          selected: false,
+          visible: false,
+          partiallyVisible: false,
+          isLeaf: false,
+          depth: 0,
+          columnsList: [],
+        },
+        metadata: {},
+        data: {},
+      };
+      const { bindings, element } = this.createColumnCellInstance(
+        this.columnElements[0]
+      );
+      bindings.bind(dummyData);
+      element.style.visibility = 'hidden';
+
+      this.columnElements[0]?.appendChild(element);
+
+      /* eslint-disable @typescript-eslint/no-explicit-any */
+      if (typeof (element as any).componentOnReady === 'function') {
+        await (element as any).componentOnReady();
+      }
+      /* eslint-enable @typescript-eslint/no-explicit-any */
+
+      let height = element.clientHeight;
+      let attempts = 0;
+
+      while (height === 0 && attempts < 10) {
+        height = await new Promise((resolve) => {
+          setTimeout(() => resolve(element.getBoundingClientRect().height), 5);
+        });
+        attempts = attempts + 1;
+      }
+      this.rowHeight = height ?? this.rowHeight;
+      element.remove();
+      this.isComputingCellHeight = false;
+    }
+  };
+
+  private computeColumnGridLayout = (): void => {
+    this.columnGridLayout = `${this.columnElements
+      .map((c) => c.initialWidth)
```

Comment:
This won't allow for a developer to programmatically change the width of a column, or query for the current width of the column. I think we can restructure a couple things to allow for this:

1. Move the header to be a child of a column.
1. Use `position: sticky` for the header element, and offset the cells by the header height.
1. Instead of grid, use flexbox to just layout columns horizontally. Resizing a column will just push over the rest of the columns.

Don't think this is a blocker for this PR. But, if it's not a huge lift would be nice to support.

Reply:
I'll take a look at this approach. Some of the column resize behavior in PLAT-27 is dependent on the grid based layout, but ideally that's not too big of a change to support